tests/api-controller-acceptance-tests/apicontroller/component_test.go

```diff
@@ -0,0 +1,397 @@
+			t.Log("not create API with wrong domain")
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName+"x", apiSecurityDisabled, true)
+
+			_, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			So(err, ShouldNotBeNil)
+		})
+
+		Convey("create API with default jwt configuration to enable authentication", func() {
+			t.Log("create API with default jwt configuration to enable authentication")
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName, apiSecurityDisabled, true)
+			authEnabled := true
+			api.Spec.AuthenticationEnabled = &authEnabled
+
+			lastApi, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			defer ctx.cleanUpApi(kymaClient, lastApi, t, false)
+			So(err, ShouldBeNil)
+			So(lastApi, ShouldNotBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+
+			lastApi, err = ctx.awaitApiCreated(kymaClient, lastApi, true, true)
+			So(err, ShouldBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+			So(lastApi.Spec, ctx.ShouldDeepEqual, api.Spec)
+
+			vs, err := istioNetClient.NetworkingV1alpha3().VirtualServices(namespace).Get(lastApi.Status.VirtualServiceStatus.Resource.Name, metav1.GetOptions{})
+			expectedVs := ctx.virtualServiceFor(testId, domainName)
+			So(err, ShouldBeNil)
+			So(vs.Spec, ctx.ShouldDeepEqual, expectedVs)
+
+			lastPolicy, err := istioAuthClient.AuthenticationV1alpha1().Policies(namespace).Get(lastApi.Status.AuthenticationStatus.Resource.Name, metav1.GetOptions{})
+			expectedPolicy := ctx.policyFor(testId, fmt.Sprintf("https://dex.%s", domainName))
+			So(err, ShouldBeNil)
+			So(lastPolicy.Spec, ctx.ShouldDeepEqual, expectedPolicy)
+		})
+
+		Convey("update API to disable authentication", func() {
+			t.Log("update API to disable authentication")
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName, apiSecurityDisabled, true)
+			authEnabled := true
+			api.Spec.AuthenticationEnabled = &authEnabled
+
+			createdApi, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			defer ctx.cleanUpApi(kymaClient, createdApi, t, false)
+			So(err, ShouldBeNil)
+			So(createdApi, ShouldNotBeNil)
+			So(createdApi.ResourceVersion, ShouldNotBeEmpty)
+
+			createdApi, err = ctx.awaitApiCreated(kymaClient, createdApi, true, true)
+			So(err, ShouldBeNil)
+			So(createdApi.ResourceVersion, ShouldNotBeEmpty)
+			So(createdApi.Spec, ctx.ShouldDeepEqual, api.Spec)
+
+			authEnabled = false
+			createdApi.Spec.AuthenticationEnabled = &authEnabled
+
+			updatedApi, err := kymaClient.GatewayV1alpha2().Apis(namespace).Update(createdApi)
+			So(err, ShouldBeNil)
+			So(updatedApi, ShouldNotBeNil)
+			So(updatedApi.ResourceVersion, ShouldNotBeEmpty)
+
+			updatedApi, err = ctx.awaitApiCreated(kymaClient, updatedApi, false, true)
```

Comment:
We should rename the function awaitApiCreated if we use it also for updates. For me, that name is misleading, as we do not wait for API creation but for API update.